src/utils.ts

```diff
@@ -181,9 +181,14 @@ export function checkMesonIsConfigured(buildDir: string) {
 }
 
 export async function rootMesonFiles(): Promise<vscode.Uri[]> {
-  const allFiles = (await vscode.workspace.findFiles("**/meson.build")).sort(
-    (a, b) => a.fsPath.length - b.fsPath.length,
-  );
+  const allFiles = (await vscode.workspace.findFiles("**/meson.build")).sort((a, b) => {
+    const aComponents = a.fsPath.split(path.sep).length;
+    const bComponents = b.fsPath.split(path.sep).length;
+
+    if (aComponents == bComponents) return a.fsPath.localeCompare(b.fsPath);
```

Comment:
I think it's a bit more complicated than this. What you want is to sort alphabetically each dir part, in pseudo-code:
```
const aComponents = a.fsPath.split(path.sep);
const bComponents = b.fsPath.split(path.sep);
len = min(aComponents.lenght,bComponents.lenght )
for (i=0; i<len; i++) {
  v = strcmp(aComponents[i], bComponents[i])
  if v!=0
    return v;
}
return aComponents.lenght - bComponents.lenght;
```

Reply:
In python that's what pathlib.Path does.

Reply:
Can you give an example where the original solution fails? I think sorting by whole path should be equivalent to sorting by each path component?

Reply:
I believe the code above would sort like that:
```
/z
/a/many/parts
/z/even/more/parts
```

But that should be sorted as
```
/a/many/parts
/z
/z/even/more/parts
```

```python
>>> sorted([Path('/z'), Path('/a/many/parts'), Path('/z/even/more/parts')])
[PosixPath('/a/many/parts'), PosixPath('/z'), PosixPath('/z/even/more/parts')]
```

Reply:
Hm, actually I take that back, because in our case we can't have both `/z` and `/z/even/more/parts`.

Reply:
But still I don't think it's logical to have `/b` before `/a` in that case:
```
/b
/a/many/parts
/z/even/more/parts
```

Reply:
It is intentional this way, to make the root-level `meson.build` the first option, or to promote the project build files against the contrib ones, e.g.

```
library/meson.build
utils/meson.build
contrib/some-lib/meson.build
```

If that's not something we want to consider, the alphabetic sort should give you the same results as sorting by path components.

```
[20:24 wolf@oberon:~]% cat a
/z
/a/many/parts
/z/even/more/parts
[20:25 wolf@oberon:~]% sort a
/a/many/parts
/z
/z/even/more/parts
```

Reply:
To be fair, I'm nitpicking, it's really a niche case after all. I guess this is good enough :)

Reply:
Ok, right, forget me. Sorry for the noise :D